include/fmt/std.h

```diff
@@ -19,18 +19,16 @@
 #ifdef __cpp_lib_filesystem
 #  include <filesystem>
 
-template <>
-struct fmt::formatter<std::filesystem::path> : formatter<string_view> {
-  template <typename FormatContext>
-  auto format(const std::filesystem::path& p, FormatContext& ctx) const ->
-      typename FormatContext::iterator {
-    return formatter<string_view>::format(p.string(), ctx);
-  }
+FMT_BEGIN_NAMESPACE
+template <typename Char>
+struct formatter<std::filesystem::path, Char> : basic_ostream_formatter<Char> {
```

Comment:
Let's not use iostreams here and do escaping with `write_escaped_string` instead.

Reply:
``write_escaped_string`` does not support ``xchar`` and does unnecessary escaping.
For std-compatible formatting of ``std::filesystem::path``, only the quote needs to be escaped.

We can implement a separate escaping function for ``std::filesystem::path``. But using iostreams allows to add support for different character types (char/wchar_t/...).

Using a separate escaping function for ``std::filesystem::path`` will allow to make implementations that do not require double memory allocation for matching character types (for conversion and escaping).  Because the ``formatter<std::filesystem::path, Char>`` has string formatting semantics, a single memory allocation for escaping will be required anyway.


@vitaut Which way to choose?

Reply:
@vitaut, ``path`` escaping is implemented in separated commit.

Reply:
I don't think it does unnecessary escaping. Note that SG16 requested the removal of path formatting from p1636 and it will likely be reintroduced in a separate paper consistent with range escaping which is what `write_escaped_string` does.

Reply:
Does this paper already exist?

Using ``write_escaped_string`` breaks compatibility with ``operator>>(std::basic_istream<CharT,Traits>& is, path& p)``. If we are ready to break the compatibility of the result with ``operator>>``, then I will rewrite the implementation to using ``write_escaped_string``.

P.S. Why ``write_escaped_string`` escape chars only for ``basic_string_view<char> ``:

https://github.com/fmtlib/fmt/blob/5c7d315ded7bdb6cc5bd65daef091eefec512f2e/include/fmt/format.h#L1692-L1693

vs

https://github.com/fmtlib/fmt/blob/5c7d315ded7bdb6cc5bd65daef091eefec512f2e/include/fmt/format.h#L1639-L1643
?

Reply:
> Does this paper already exist?

No.

Compatibility with `operator>>` is not critical.

> Why write_escaped_string escape chars only for basic_string_view<char> ... ?

It hasn't been implemented yet for other types.

Reply:
I added ``xchar`` support for ``write_escape_string`` in a separate PR #2904.

After merging #2904, I will make changes to this PR.

Reply:
Done